src/components/TodosList.jsx

```diff
@@ -1,20 +1,23 @@
 import React, { useState } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faPenToSquare, faTrash, faCheck, faXmark } from '@fortawesome/free-solid-svg-icons';
+import { data } from '../staticData/buttonData';
 
 const TodosList = ({ todos, setTodos, editTodo, setEditTodo, isEdit, setIsEdit, errorMessage, setErrorMessage }) => {
   const [editedText, setEditedText] = useState('');
-  const [filter, setFilter] = useState('all');
+  const [filter, setFilter] = useState('All'); // this is for filter the todos
+  const [active, setActive] = useState(null)   // this is for change the color of the buttons on click
 
-  const handleFilter = (filterType) => {
-    setFilter(filterType);
+  const handleFilter = (id,name) => {
+    setFilter(name)
+    setActive(id)
   };
 
   let filteredTodos = todos;
 
-  if (filter === 'completed') {
+  if (filter === 'Completed') {
     filteredTodos = todos.filter(todo => todo.completed);
-  } else if (filter === 'incomplete') {
+  } else if (filter === 'Incomplete') {
     filteredTodos = todos.filter(todo => !todo.completed);
   }
 
```

Comment:
this is not way to do this.
it will run on every any state change .
do in useEffect 
declare variable in useState